### PR TITLE
feat: AWS Bedrock Claude3

### PR DESCRIPTION
**Pull request by soulteary**
# Description

Support calling AWS Bedrock Claude3 model, issue #2856

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

image

image

image

image

image

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods
- [x] `optional` I have made corresponding changes to the documentation 
- [x] `optional` I have added tests that prove my fix is effective or that my feature works
- [x] `optional` New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by guchenhe** [comment]
CleanShot 2024-03-17 at 01 31 54@2x

Thanks for the PR. I wasn't able to get either Claude 3 Sonnet or Haiku to work due to the above error msg. Would you take a look into it?

**Comment by soulteary** [comment]
> CleanShot 2024-03-17 at 01 31 54@2x
> 
> Thanks for the PR. I wasn't able to get either Claude 3 Sonnet or Haiku to work due to the above error msg. Would you take a look into it?

@guchenhe This bug comes from Claude SDK. Refer to the requirement.txt I submitted. You need to update an SDK version to the latest one.

I recommend verifying directly in a fresh Docker environment, it's very easy.

File: api/core/model_runtime/model_providers/bedrock/llm/anthropic.claude-3-haiku-v1.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+model: anthropic.claude-3-haiku-20240307-v1:0
+label:
+  en_US: Claude3 Haiku
```
Comment (guchenhe) [inline api/core/model_runtime/model_providers/bedrock/llm/anthropic.claude-3-haiku-v1.yaml]:
```suggestion
  en_US: Claude 3 Haiku
```

File: api/core/model_runtime/model_providers/bedrock/llm/anthropic.claude-3-haiku-v1.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      zh_Hans: 停止前生成的最大令牌数。请注意，Anthropic Claude 模型可能会在达到 max_tokens 的值之前停止生成令牌。不同的 Anthropic Claude 模型对此参数具有不同的最大值。
+      en_US: The maximum number of tokens to generate before stopping. Note that Anthropic Claude models might stop generating tokens before reaching the value of max_tokens. Different Anthropic Claude models have different maximum values for this parameter.
+  # docs: https://docs.anthropic.com/claude/docs/system-prompts
+  - name: system
+    type: string
+    required: false
+    default: "Respond only in English."
+    help:
+      zh_Hans: 系统提示请求。系统提示是向 Anthropic Claude 提供上下文和指令的一种方式，例如指定特定目标或角色。
+      en_US: The system prompt for the request. A system prompt is a way of providing context and instructions to Anthropic Claude, such as specifying a particular goal or role.
```
Comment (guchenhe) [inline api/core/model_runtime/model_providers/bedrock/llm/anthropic.claude-3-haiku-v1.yaml]: This seems the same as system prompt in expert mode. Can it be removed from both yamls?

File: api/core/model_runtime/model_providers/bedrock/llm/anthropic.claude-3-sonnet-v1.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
+model: anthropic.claude-3-sonnet-20240229-v1:0
+label:
+  en_US: Claude3 Sonnet
```
Comment (guchenhe) [inline api/core/model_runtime/model_providers/bedrock/llm/anthropic.claude-3-sonnet-v1.yaml]:
```suggestion
  en_US: Claude 3 Sonnet
```

**Review by guchenhe — CHANGES_REQUESTED** [review]

**Comment by guchenhe** [comment]
> > CleanShot 2024-03-17 at 01 31 54@2x
> > Thanks for the PR. I wasn't able to get either Claude 3 Sonnet or Haiku to work due to the above error msg. Would you take a look into it?
> 
> @guchenhe This bug comes from Claude SDK. Refer to the requirement.txt I submitted. You need to update an SDK version to the latest one.
> 
> I recommend verifying directly in a fresh Docker environment, it's very easy.

Thanks! It's working like a charm now. I left some suggestions and we'll be able to merge as soon as those are resolved and linting passes

**Review by crazywoola — DISMISSED** [review]

**Review by crazywoola — APPROVED** [review]

**Review by takatost — APPROVED** [review]
LGTM

**Pull request merged**